sys/freebsd/sys.txt

```diff
@@ -136,6 +137,7 @@ copy_file_range(infd fd, inoffp ptr[inout, fileoff[int64]], outfd fd, outoffp pt
 __realpathat(dirfd fd, path ptr[in, filename], buf ptr[out, string], size len[buf], flags flags[realpathat_flags])
 undelete(path ptr[in, filename])
 acct(filename ptr[in, filename, opt])
+__specialfd(type flags[specialfd_type], req buffer[in], len len[req])
```

Comment:
Let's specialize this. Each specialfd type takes a type-specific structure. See `lib/libc/gen/eventfd.c` for examples. So I would suggest defining event fds this way:

```
eventfd {
    initval int32
    flags int32
}
__specialfd$eventfd(type const[SPECIALFD_EVENTFD], req ptr[in, eventfd], len len[req])
```

Also, __specialfd returns a new fd which can be passed to other system calls, so it should be annotated as returning an fd.